### PR TITLE
[BLADE-202] Merge Child Predicates

### PR DESCRIPTION
**Pull request by goran-ethernal**
This PR does the following:

- merges Child predicates so we don't have `Child` and `ChildMintable` contracts.
- removes addresses of `StateReceiver` and `ExitHelper` contracts from predicate contracts, since all the logic now goes through `Gateway` contract.
- updates tests.

### CODE REVIEW

File: contracts/blade/ChildERC1155Predicate.sol [hunk 1]
```diff
@@ -6,23 +6,17 @@ import "@openzeppelin/contracts/proxy/Clones.sol";
...
-    /// @custom:security write-protection="onlySystemCall()"
+contract ChildERC1155Predicate is IChildERC1155Predicate, Initializable {
     IGateway public gateway;
-    /// @custom:security write-protection="onlySystemCall()"
-    address public stateReceiver;
-    /// @custom:security write-protection="onlySystemCall()"
     address public rootERC1155Predicate;
-    /// @custom:security write-protection="onlySystemCall()"
-    address public childTokenTemplate;
+    address public sourceTokenTemplate;
```
Comment (Stefan-Ethernal) [inline contracts/blade/ChildERC1155Predicate.sol]:
```suggestion
    address public destinationTokenTemplate;
```

File: contracts/blade/ChildERC1155Predicate.sol [hunk 2]
```diff
@@ -72,19 +66,17 @@ contract ChildERC1155Predicate is IChildERC1155Predicate, Initializable, System
...
      * @dev Can only be called once.
      */
     function initialize(
         address newGateway,
-        address newStateReceiver,
         address newRootERC1155Predicate,
-        address newChildTokenTemplate
-    ) public virtual onlySystemCall initializer {
-        _initialize(newGateway, newStateReceiver, newRootERC1155Predicate, newChildTokenTemplate);
+        address newSourceTokenTemplate
```
Comment (Stefan-Ethernal) [inline contracts/blade/ChildERC1155Predicate.sol]:
```suggestion
        address newDestinationTokenTemplate
```

File: contracts/blade/ChildERC1155Predicate.sol (hunk 1 above)
Comment (goran-ethernal) [reply contracts/blade/ChildERC1155Predicate.sol]: Fix: https://github.com/Ethernal-Tech/blade-contracts/pull/24/commits/82d90d486467ac8e6bb9f8fc4645ad93b6d88eaf

File: contracts/blade/ChildERC1155Predicate.sol (hunk 2 above)
Comment (goran-ethernal) [reply contracts/blade/ChildERC1155Predicate.sol]: Fix: https://github.com/Ethernal-Tech/blade-contracts/pull/24/commits/82d90d486467ac8e6bb9f8fc4645ad93b6d88eaf

File: contracts/interfaces/bridge/IStateReceiver.sol [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+interface IStateReceiver {
```
Comment (Stefan-Ethernal) [inline contracts/interfaces/bridge/IStateReceiver.sol:4]:
```suggestion
interface IReceiver {
```

File: contracts/interfaces/bridge/IStateReceiver.sol [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    /**
+     * @notice Called by gateway when state is received from source chain
+     * @param sender Address of the sender on the child chain
+     * @param data Data sent by the sender
+     */
+    function onStateReceive(uint256 id, address sender, bytes calldata data) external;
```
Comment (Stefan-Ethernal) [inline contracts/interfaces/bridge/IStateReceiver.sol:10]: WDYT (just don't forget to change the string under quotas in the Gateway, where this function gets invoked).
```suggestion
    function onMsgReceived(uint256 id, address sender, bytes calldata data) external;
```

**Review by Stefan-Ethernal — APPROVED** [review]

**Review by dusannosovic-ethernal — APPROVED** [review]

File: contracts/interfaces/bridge/IStateReceiver.sol (hunk 3 above)
Comment (goran-ethernal) [reply contracts/interfaces/bridge/IStateReceiver.sol:4]: We will address this in another task.

File: contracts/interfaces/bridge/IStateReceiver.sol (hunk 4 above)
Comment (goran-ethernal) [reply contracts/interfaces/bridge/IStateReceiver.sol:10]: We will address this in another task.

**Pull request merged**